.github/config-schema.json

```
@@ -338,7 +338,14 @@
         "read_only_style": "red",
         "repo_root_format": "[$before_root_path]($style)[$repo_root]($repo_root_style)[$path]($style)[$read_only]($read_only_style) ",
         "repo_root_style": null,
-        "style": "cyan bold",
+        "style": [
+          {
+            "env": null,
+            "expected_value": null,
+            "operator": null,
```

```suggestion
            "when": null,
```

---

I think it would be better to do this. See my other other comment above.

---

@rogeruiz If I understand correctly you are suggesting to evaluate shell command and apply style accordingly basing on the command output? (similarly to what custom module does) 

The solution you're proposing is in fact simpler, however it is not platform independent. Starship runs on many different shells and for each of them a script can have different syntax. Thus starship's configurations using conditional-style, won't be applicable for all setups.  For example a simple `when` script to compare a value of some env variable with some string is implemented differently on for `bash` and `powershell`

---

Hey @cymruu good call on that. I think that could be solved with the way custom commands are implemented. Because this seems like a custom style based on some external thing being true or not. 

[=> custom command docs](https://starship.rs/config/#custom-commands)

From my understanding then maybe it should be up to the user to define all these caveats? Perhaps the non-string version of style should take a map of the `os`, `when`, and the `format`?

I know that seems complex, but I think the level of customization you're looking looking to achieve here should require that many granular controls. 

Perhaps something like this? It feels like a lot of repetition and I agree that `when` commands aren't portable.

```suggestion
    {
        os = "linux",
        when = """ true """,
        custom_style = "bold dimmed"
    },
    {
        os = "windows",
        when = """ true """,
        custom_style = "bold dimmed"
    },
```

---

Also sorry that patch won't resolve properly. I was trying to get my point across quickly. I will need to take a look at the branch locally to see if what I'm saying makes sense.